### ISSUE TITLE
Django templates: recreate logic for initial loading of per-page critical assets

### ISSUE THREAD

**Issue by ryanpitts** [OpenNews/opennews-source#12]
I think this is going to need to be done on an app-by-app basis, in the `_base_APPNAME.html` templates. For now, will let the primary base template just call for `main.css` and plan to come back to this later.

**Comment by ryanpitts** [comment]
For reference, the PHP to recreate:

```
<?php
if ( $_COOKIE[ "fullCSS" ] == "loaded" || $_GET[ "static" ] == "true" ) {
?>
<link href="/dist/css/main.css" rel="stylesheet" />
<?php
} else {
    if ( isset( $criticalcss ) ) {
?>
<style><?php include( $rootpath . "/dist/css/" . $criticalcss ); ?></style>
<?php
    } else {
?>
<link href="/dist/css/main.css" rel="stylesheet" />
<?php
    }
}
?>
```

**Comment by ryanpitts** [comment]
@beep this approach is purely for performance, not for splitting off features into separate js files, right?

**Comment by beep** [comment]
@ryanpitts Yeah, exactly! The full logic’s outlined in a few places online ([here’s a good, if long, post on the subject](https://www.filamentgroup.com/lab/performance-rwd.html)), but the TLDR is:

1. On first page view, we load the “critical” CSS inline, which prevents the CSS from blocking rendering.
    - The critical CSS is basically everything required for viewing the “above the fold” view of the design.
1. In the background, the full CSS file (`main.css`) is loaded via JavaScript, and then a `fullCSS` cookie is set.
1. On subsequent page views, if the cookie is present, the template just uses the `link`’d CSS file, which is now cached.

Does that make sense? Any questions or concerns with that approach?

**Comment by ryanpitts** [comment]
Nope, seems great. For now I'm just skipping straight to `main.css` in the interest of getting templates integrated: https://github.com/OpenNews/opennews-source/blob/master/source/base/templates/_v2/base.html#L26

Will come back to this once when I can, so just checking to make sure that the stuff in `main` is a superset of the stuff in `critical`.

**Comment by beep** [comment]
Oh, that is supremely sensible and smart! Let’s definitely circle back on it later. (And if we need to rip it out for launch, we can toooootally do that.)
